pico/tools/src/main/java/io/helidon/pico/tools/DefaultInterceptorCreator.java

```diff
@@ -424,15 +426,30 @@ public Set<String> allAnnotationTypeTriggers() {
 
         @Override
         public Optional<InterceptionPlan> createInterceptorPlan(Set<String> interceptorAnnotationTriggers) {
-            List<InterceptedElement> interceptedElements = getInterceptedElements(interceptorAnnotationTriggers);
-            if (interceptedElements == null || interceptedElements.isEmpty()) {
-                return Optional.empty();
+            boolean hasNoArgConstructor = hasNoArgConstructor();
+            Set<TypeName> interfaces = interfaces();
+
+            // the code generation will extend the class when there is a zero/no-arg constructor, but if not then we will use a
```

Comment:
- we can always extend the class, and this should always work as long as the instance is created by Pico (as we can inject constructor parameters into subclass constructor and call `super()`)
- we should never create a new instance (even if it has a no-arg constructor) that will not be used as "the" service instance (as any constructor may have side effects that we would trigger outside of user's expectations)
- the only case where we should only support interface injections is the case where the underlying instance gets created by a producer (and not by Pico)

Reply:
I think this whole section is not correct and not what we agreed on.

Reply:
discussing over slack

Reply:
Added to https://github.com/helidon-io/helidon/issues/6542